code/_onclick/click.dm

```diff
@@ -213,11 +213,13 @@
 		for(var/atom/target in checking)  // will filter out nulls
 			if(closed[target] || isarea(target))  // avoid infinity situations
 				continue
-			closed[target] = TRUE
+
 			if(isturf(target) || isturf(target.loc) || (target in direct_access) || (ismovable(target) && target.flags_1 & IS_ONTOP_1)) //Directly accessible atoms
 				if(Adjacent(target) || (tool && CheckToolReach(src, target, tool.reach))) //Adjacent or reaching attacks
 					return TRUE
 
+			closed[target] = TRUE
+
```

Comment:
microest of micro opts since closed isnt accessed at all inside those if statements but they can return so by putting it after the check you have a chance of not needing to do the associative write

Reply:
dirty, disgusting, impressive